lib/access-context.js

```diff
@@ -126,8 +128,20 @@ AccessContext.prototype.addPrincipal = function(principalType, principalId, prin
 AccessContext.prototype.getUserId = function() {
   for (var i = 0; i < this.principals.length; i++) {
     var p = this.principals[i];
+    // the principalType must either be 'USER' or permit to resolve a valid user model
     if (p.type === Principal.USER) {
       return p.id;
+    } else {
+      if (!this.registry) return null;
```

Comment:
about asserting this.registry in the constructor: yes, good idea
about using accessContext.model.registry as a fallback, we need to be cautious about that, there's currently a dangerous
'model = ('string' === typeof model) ? loopback.getModel(model) : model; '
in the constructor, which will bring the model from the global registry, and thus we'd fetch the global registry instead of the local one
I'll manage something. Thoughts?